meilisearch-release-changelog/v1.2.0.md

```diff
@@ -0,0 +1,203 @@
+  - The number of typo considered for a single term is now limited to 150 for 1 typo, 50 for 2 typos. Similarly, the number of considered (single word or multi word) synonyms is limited to 50, the total number of words for all synonyms of a term is limited to 100.
+  - The number of words considered in a query is now limited to 1000. The number of terms (words not inside a phrase, and individual phrases) was and is still limited to 10, so this change only impacts queries containing very long phrases.
+- Geo search improvements
+  - Improved performance when there are only a few documents to geosort.
+  - Descending sort is now as performant as ascending sort.
+- Solves some performance issues: https://github.com/meilisearch/meilisearch/issues/3124, https://github.com/meilisearch/meilisearch/issues/3378, https://github.com/meilisearch/meilisearch/issues/2202, https://github.com/meilisearch/meilisearch/issues/3123
+
+### Relevancy improvements
+
+- Synonyms are no longer considered as exact matches by the `exactness` ranking rule. This change is due to the goal of the `exactness` ranking rule of boosting the documents that contain the query exactly as typed by the user. Other ranking rules will still consider synonyms. Solves https://github.com/meilisearch/meilisearch/issues/3410
+- `Attributes`, `Exactness`, `typo` and `proximity` ranking rules now imply `words` if `words` was not already specified before them. This change was made because the behavior of applying these ranking rules before or without `words` was unintuitive and could hurt relevancy.
+- Split words are now looked for for digrams. This means that the query `whit ehorse` may now match documents containing `white horse`, which improves the relevancy of typo tolerance.
+- ngrams and split words are now disadvantaged in comparison to the exact word in the `typo` ranking rule.
+- The "iterative" version of ranking rules are now removed (https://github.com/meilisearch/meilisearch/issues/3356). The iterative version of algorithms would be applied depending on internal factors and would sometimes differ greatly in relevancy from the base ranking rule, leading to inconsistent relevancy.
+- Exactness will not prioritize documents containing exact words which were previously removed by the `words` ranking rule anymore. For example, when using the query `Batman: the dark knight rises` with the default term matching strategy (`last`), documents with a title starting with `Batman: the dark knight` will now be returned before documents that contain additional terms like `Batman: the dark knight crashes`, which improves relevancy. Congratulations if you read so far. Similarly, with the query: `Batman the supercalifragilisticexpialidocious knight rises`. Then the `Last` term matching strategy will reduce the query to `Batman the`. Exactness will then only prioritise documents that contain the exact words `Batman` and `the` but not `supercalifragilisticexpialidocious`, `knight`, or `rises`.
```

Comment:
Are the explanation from `Congratulations...` necessary? The first explanation is not complete but maybe enough?

Reply:
Agreed. It might be just me being tired, but the final explanation actually confused me more than it helped me understand this change.

```suggestion
- Exactness will no longer prioritize documents containing exact words which have already been parsed and removed by the `words` ranking rule. For example, when using the query `Batman: the dark knight rises` with the default term matching strategy (`last`), documents with a title starting with `Batman: the dark knight` will be returned before documents containing additional terms like `Batman: the dark knight crashes`
```